### ISSUE TITLE
Different behavior from ExtraSpacing when used with and without --only flag

### ISSUE THREAD

**Issue by backus** [bbatsov/rubocop#3953]
Noticed this while looking into #3945. Not sure how many cops this applies to.

### Normal invocation

```sh
$ echo "foo   { bar }\nthing { bar }" | bin/rubocop --stdin -
Inspecting 1 file
C

Offenses:

-:1:1: C: The name of this source file (-) should use snake_case.
foo   { bar }
^
-:1:1: C: Missing frozen string literal comment.
foo   { bar }
^
-:1:4: C: Unnecessary spacing detected.
foo   { bar }
   ^^

1 file inspected, 3 offenses detected
```

### Using `--only ExtraSpacing`

```sh
$ echo "foo   { bar }\nthing { bar }" | bin/rubocop --only ExtraSpacing --stdin -
Inspecting 1 file
.

1 file inspected, no offenses detected
```

Latest version of rubocop:

```sh
$ bin/rubocop -V
0.47.1 (using Parser 2.3.3.1, running on ruby 2.3.1 x86_64-darwin16)

$ git pull
Already up-to-date.
```

**Comment by backus** [comment]
Ah sigh I had a local change still in the mix... Sorry for the noise
